fetch.go

```diff
@@ -74,6 +74,10 @@ type Fetcher struct {
 	// this is a breaking change, won't fix.
 	HttpClient Doer
 
+	// The Doer to use for a specific command. If nil, falls back to HttpClient
+	// field, and then the net/http default client.
+	HTTPClientFunc func(cmd Command) Doer
```

Comment:
I named the field `HTTPClientFunc` in order to stay consistent with the `HttpClient` field, but a more accurate name might be something like `DoerFunc`, since there is no requirement that the returned object is a `*http.Client`.